lib/msf/core/post/file.rb

```diff
@@ -578,6 +579,32 @@ def copy_file(src_file, dst_file)
 
 protected
 
+  def _read_file_powershell(filename)
+    eof = cmd_exec("[System.IO.File]::ReadAllBytes(\"#{filename}\").Length").strip.to_i
```

Comment:
It should be possible to remove this which would make things faster when the file length is not a multiple of the chunk size, which for a large chunk size is unlikely.

What you would do is update `_read_file_powershell_fragment` to take the chunksize and offset (calculate the stop using `start + chunk_size - 1`). Then from this function, when the size of the chunk that is returned is less than the size you requested, you know that the whole file has been read. If it's a multiple, you'll end up reading it one time more than is necessary but that's better than *always* retrieving the file length. You'll no longer have to track `eof` or `stop`.

The peudo code would be something like this:
```ruby
data = ''
offset = 0
chunk_size = 1024 * 1024
loop do
  chunk = _read_file_powershell_fragment(filename, chunk_size, offset)
  data << chunk
  break if chunk.length < chunk_size
end
```

Note that uses the suggested signature for `_read_file_powershell_fragment`.

Reply:
okay thanks, will remove that check :+1: